captum/optim/_param/image/images.py

```diff
@@ -1,538 +1,446 @@
-        else:
-            if squash_func is None:
-                squash_func: Callable[
-                    [torch.Tensor], torch.Tensor
-                ] = lambda x: torch.sigmoid(x)
-        self.squash_func = squash_func
-        self.parameterization = parameterization(
-            size=size, channels=channels, batch=batch, init=init
-        )
-
-    def forward(self) -> torch.Tensor:
-        image = self.parameterization()
-        if self.decorrelate is not None:
-            image = self.decorrelate(image)
-        image = image.rename(None)  # TODO: the world is not yet ready
-        return CudaImageTensor(self.squash_func(image))
+from copy import deepcopy
+from typing import Callable, List, Optional, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+try:
+    from PIL import Image
+except (ImportError, AssertionError):
+    print("The Pillow/PIL library is required to use Captum's Optim library")
+
+from captum.optim._param.image.transform import SymmetricPadding, ToRGB
+
+
+class ImageTensor(torch.Tensor):
+    @classmethod
+    def open(cls, path: str, scale: float = 255.0):
+        if path.startswith("https://") or path.startswith("http://"):
+            response = requests.get(path, stream=True)
+            img = Image.open(response.raw)
+        else:
+            img = Image.open(path)
+        img_np = np.array(img.convert("RGB")).astype(np.float32)
+        return cls(img_np.transpose(2, 0, 1) / scale)
+
+    def __repr__(self) -> str:
+        prefix = "ImageTensor("
+        indent = len(prefix)
+        tensor_str = torch._tensor_str._tensor_str(self, indent)
+        suffixes = []
+        if self.device.type != torch._C._get_default_device() or (
+            self.device.type == "cuda"
+            and torch.cuda.current_device() != self.device.index
+        ):
+            suffixes.append("device='" + str(self.device) + "'")
+        return torch._tensor_str._add_suffixes(
+            prefix + tensor_str, suffixes, indent, force_newline=self.is_sparse
+        )
+
+    def show(self, scale: float = 255.0) -> None:
+        if len(self.shape) == 3:
+            numpy_thing = self.cpu().detach().numpy().transpose(1, 2, 0) * scale
+        elif len(self.shape) == 4:
+            numpy_thing = self.cpu().detach().numpy()[0].transpose(1, 2, 0) * scale
+        plt.imshow(numpy_thing.astype(np.uint8))
+        plt.axis("off")
+        plt.show()
+
+    def export(self, filename: str, scale: float = 255.0) -> None:
+        colorspace = "RGB" if self.size(1) == 3 else "RGBA"
+        if len(self.shape) == 3:
+            numpy_thing = self.cpu().detach().numpy().transpose(1, 2, 0) * scale
+        elif len(self.shape) == 4:
+            numpy_thing = self.cpu().detach().numpy()[0].transpose(1, 2, 0) * scale
+        im = Image.fromarray(numpy_thing.astype("uint8"), colorspace)
+        im.save(filename)
```

Comment:
So, I was looking at the code, and it doesn't seem like the current ImageTensor works with CUDA tensors:

```
class ImageTensor(torch.Tensor):
     ...

x = torch.ones(1,3,4,4).cuda()
x_out = ImageTensor(x)

# TypeError: expected CPU (got CUDA)
```